hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+                        shp_full_path=target
+                    files.append(UploadedFile(file=open(target, 'r'), name=fileName + fileExtension))
+                    uploadedFileCount+=1
+                    uploadedFilenameString_dict[fileName + fileExtension]=str(1)
+                else:
+                    dir_count+=1 #folder count +1 (should be 0 or 1 folder)
+
+            rslt['shp_full_path']=shp_full_path
+
+            rslt["baseFilename"]=baseFilename
+            rslt["uploadedFileCount"]=uploadedFileCount
+            uploadedFilenameString = json.dumps(uploadedFilenameString_dict)
+            rslt["uploadedFilenameString"]=uploadedFilenameString
+            rslt["are_files_valid"] = True
+            rslt['message'] = 'All files are validated.'
+        else:
+            rslt["are_files_valid"] = False
+            rslt['message'] = 'Please upload valid file(s).'
+
+        return rslt
+
+def parse_shp_zshp(uploaded_file_type, baseFilename, uploadedFileCount, uploadedFilenameString, shp_full_path):
+    try:
+        metadata_array=[]
+        metadata_dict={}
+
+        #fileTypeInfo_dict
+        originalFileInfo_dict={}
+        originalFileInfo_dict["fileType"]="SHP" if uploaded_file_type == "shp" else "ZSHP"
+        originalFileInfo_dict["baseFilename"] = baseFilename
+        originalFileInfo_dict["fileCount"]= uploadedFileCount
+        originalFileInfo_dict["filenameString"]=uploadedFilenameString
+        metadata_array.append({"OriginalFileInfo": originalFileInfo_dict})
+        metadata_dict["originalfileinfo"]=originalFileInfo_dict
+
+        # wgs84 extent
+        parsed_md_dict = parse_shp(shp_full_path)
+        if parsed_md_dict["wgs84_extent_dict"]["westlimit"]!=UNKNOWN_STR:
+            coverage_dict={"Coverage": {"type": "box", "value": parsed_md_dict["wgs84_extent_dict"]}}
+            metadata_array.append(coverage_dict)
+            metadata_dict["coverage"]=coverage_dict
+
+        #original extent
+        original_coverage_dict ={}
+        original_coverage_dict["originalcoverage"]={"northlimit":parsed_md_dict["origin_extent_dict"]["northlimit"],
+                                                    "southlimit":parsed_md_dict["origin_extent_dict"]["southlimit"],
+                                                    "westlimit":parsed_md_dict["origin_extent_dict"]["westlimit"],
+                                                    "eastlimit":parsed_md_dict["origin_extent_dict"]["eastlimit"],
+                                                    "projection_string":parsed_md_dict["origin_projection_string"],
+                                                    "projection_name":parsed_md_dict["origin_projection_name"],
+                                                    "datum":parsed_md_dict["origin_datum"],
+                                                    "unit":parsed_md_dict["origin_unit"]
+                                                    }
```

Comment:
this dictionary could be cleaned up with spacing and less indentation to be read more easily